ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -58,14 +67,22 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
         new DropTopicCommand(dataSource.getKsqlTopic().getTopicName());
     metaStore.deleteSource(sourceName);
     dropTopicCommand.run(metaStore, isValidatePhase);
-    if (!isValidatePhase && dataSource.getKsqlTopic().getKsqlTopicSerDe().getSerDe()
-                            == DataSource.DataSourceSerDe.AVRO) {
+    if (!isValidatePhase && withTopic) {
```

Comment:
readability would be better if this block was extracted into a method

Reply:
Moved it into a method.